geopandas/io/sql.py

```diff
@@ -98,3 +99,277 @@ def load_geom_text(x):
+    chunks = np.array_split(gdf, chunk_cnt)
+    return chunks
+
+
+def _write_to_db(gdf, engine, index, tbl, srid, geom_name, if_exists, chunksize):
+
+    conn = engine.raw_connection()
+    cur = conn.cursor()
+
+    try:
+        # If appending to an existing table, temporarily change
+        # the srid to 0, and update the SRID afterwards
+        if if_exists == "append":
+            sql = "SELECT UpdateGeometrySRID('{schema}','{tbl}','{geom}',{crs})".format(
+                schema=tbl.table.schema, tbl=tbl.table.name, geom=geom_name, crs=0
+            )
+            cur.execute(sql)
+
+        if chunksize is None:
+            _populate_db(gdf, conn, cur, index, tbl)
+        else:
+            # Insert in chunks
+            chunks = _get_chunks(gdf, chunksize)
+
+            for chunk in chunks:
+                _populate_db(chunk, conn, cur, index, tbl)
+
+        # SRID needs to be updated afterwards as Shapely does not support
+        # EWKT/EWKB geometries, see:
+        # https://community.gispython.narkive.com/qTVQCl3f/ewkt-ewkb-support-in-shapely
+        sql = "SELECT UpdateGeometrySRID('{schema}','{tbl}','{geom}',{srid})".format(
+            schema=tbl.table.schema, tbl=tbl.table.name, geom=geom_name, srid=srid
+        )
+        cur.execute(sql)
+        conn.commit()
+
+    except Exception as e:
+        conn.connection.rollback()
+        raise e
+    finally:
+        conn.close()
+
+
+def write_postgis(
+    gdf,
+    name,
+    con,
+    schema=None,
+    if_exists="fail",
+    index=False,
+    index_label=None,
+    chunksize=None,
+    dtype=None,
+):
+    """
+    Upload GeoDataFrame into PostGIS database.
+
+    Parameters
+    ----------
+    name : str
+        Name of the target table.
+    con : sqlalchemy.engine.Engine
+        Active connection to the PostGIS database.
+    if_exists : {‘fail’, ‘replace’, ‘append’}, default ‘fail’
+        How to behave if the table already exists.
+          - fail: Raise a ValueError.
+          - replace: Drop the table before inserting new values.
+          - append: Insert new values to the existing table.
+    schema : string, optional
+        Specify the schema. If None, use default schema: 'public'.
+    index : bool, default True
+        Write DataFrame index as a column.
+        Uses *index_label* as the column name in the table.
+    index_label : string or sequence, default None
+        Column label for index column(s).
+        If None is given (default) and index is True,
+        then the index names are used.
+    chunksize : int, optional
+        Rows will be written in batches of this size at a time.
+        By default, all rows will be written at once.
+    dtype : dict of column name to SQL type, default None
+        Specifying the datatype for columns.
+        The keys should be the column names and the values
+        should be the SQLAlchemy types.
+    """
+    from geoalchemy2 import Geometry
```

Comment:
Done 👍

Reply:
Thanks @jorisvandenbossche for pointing out the possibility to create EWKB in shapely! :) Did some searches back in the days about this, but for some reason didn't ever find this issue about the topic. Hmm, I think it would definitely be a "cleaner" solution to create EWKB instead of doing these sql tricks with the geometry column. But let's check the influence on the performance and whether this also fixes the "append" method. I do some tests about these things.

Reply:
@jorisvandenbossche Okay, did some tests and it seems that creating the EWKB takes twice as long compared to WKB without the srid information with shapely:

Creating 10 000 point wkbs without srid:
`116 ms ± 2.24 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)`

Creating 10 000 point ewkbs:
`226 ms ± 711 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)`

In case of updating the srid of pygeos array, I guess the cost is very slow as things are updated in a vectorized manner.  

Creating the EWKB geometries allows us to drop all of the SQL tricks related to updating the SRID of the geometry column. Hence, I would vote for accepting this cost increase in terms of shapely operations as the codes regarding the `to_postgis()` would become much leaner.